packages/manager/src/views/WebPropertyEnvPage/components/EditAccess/EditMemberAccess.tsx

```diff
@@ -0,0 +1,206 @@
+import { useAddPermission, useDeleteMember } from '@app/services/rbac';
+import { toPascalCase } from '@app/utils/toPascalConvert';
+import {
+  Accordion,
+  AccordionContent,
+  AccordionItem,
+  AccordionToggle,
+  Button,
+  Checkbox,
+  Split,
+  SplitItem
+} from '@patternfly/react-core';
+import { TableComposable, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+import React, { useState } from 'react';
+import toast from 'react-hot-toast';
+interface Props {
+  onClose: () => void;
+  editMemberName: string;
+  propertyIdentifier: string;
+  memberList: any;
+}
+type ColumnNames = {
+  NAME: string;
+  APIKEY_CREATION: string;
+  APIKEY_DELETION: string;
+  PERMISSION_CREATION: string;
+  PERMISSION_DELETION: string;
+  ENV_CREATION: string;
+  ENV_SYNC: string;
+  [key: string]: string;
+};
+type GroupItem1 = {
+  email: string;
+  name: string;
+  role: string;
+  PERMISSION_CREATION: boolean;
+  PERMISSION_DELETION: boolean;
+  ENV_SYNC: boolean;
+  ENV_CREATION: boolean;
+  APIKEY_DELETION: boolean;
+  APIKEY_CREATION: boolean;
+};
+export const EditMemberAccess = ({
+  onClose,
+  editMemberName,
+  propertyIdentifier,
+  memberList
+}: Props): JSX.Element => {
+  const [expanded, setExpanded] = useState('def-list-toggle2');
+  const [addAccess, setAddAccess] = useState<string[]>([]);
+  const [deleteAccess, setDeleteAccess] = useState<string[]>([]);
+  const useAddPermission1 = useAddPermission(propertyIdentifier);
+  const deleteMember = useDeleteMember(propertyIdentifier);
+  const data1 = memberList.data.filter((e: GroupItem1) => e.name === editMemberName);
+  const [group, setGroup] = useState({ data: data1 });
+  const columnNames: ColumnNames = {
+    NAME: 'Name',
+    // ROLE: 'Role',
+    APIKEY_CREATION: 'APIKEY_CREATION',
+    APIKEY_DELETION: 'APIKEY_DELETION',
+    PERMISSION_CREATION: 'PERMISSION_CREATION',
+    PERMISSION_DELETION: 'PERMISSION_DELETION',
+    ENV_CREATION: 'ENV_CREATION',
+    ENV_SYNC: 'ENV_SYNC'
+  };
+  const onToggleAccordian = async (id: string) => {
+    if (id === expanded) {
+      setExpanded('');
+    } else {
+      setExpanded(id);
+    }
+  };
+  const handleChange = (
+    checked: boolean,
+    event: React.FormEvent<HTMLInputElement>,
+    email: string
+  ) => {
+    const target = event.currentTarget as HTMLInputElement;
+    const temp = group.data;
+    const objIndex = temp.findIndex((e: GroupItem1) => e.email === email);
+    temp[objIndex][target.name] = checked;
```

Comment:
How about we just do it one line

```js
setGroup(({data})=>({data:data.map((e)=>email===e.email?{...e,[target.name]:checked}:e)}))
```

Reply:
Explanation: Map the array and if the email is one we want we change the key target.name else keep it as it is

Reply:
done